### PR TITLE
fixes #8880 yum plugins repo, removed rc

### PR DESCRIPTION
**Pull request by endyman**
This PR adds yum plugins repo and removes (according @domcleal) outdated rc versions and tests.

### CODE REVIEW

File: manifests/install/repos/yum.pp [hunk 1]
```diff
@@ -1,8 +1,13 @@
 # Install a yum repo
 define foreman::install::repos::yum ($repo, $yumcode, $gpgcheck) {
   $repo_path = $repo ? {
-    'stable' => 'releases/latest',
-    default  => $repo,
+    'stable'  => 'releases/latest',
+    'nightly' => $repo,
+    default   => "releases/${repo}",
```
Comment (domcleal) [inline manifests/install/repos/yum.pp]: We might need to handle the case where somebody's already passing "releases/1.7" or similar in, so we don't prepend "releases/" again.

File: manifests/init.pp [hunk 2]
```diff
@@ -35,7 +35,7 @@
 #                             set it to some custom location.
 #                             type:boolean
 #
-# $repo::                     This can be stable, rc, or nightly
+# $repo::                     This can be stable, nightly or a apecific version i.e. 1.7
```
Comment (domcleal) [inline manifests/init.pp]: specific

**Comment by mmoll** [comment]
:+1:

File: manifests/install/repos/yum.pp [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
-    'stable' => 'releases/latest',
-    default  => $repo,
+    'stable'      => 'releases/latest',
+    /^releases\// => $repo,
+    'nightly'     => $repo,
+    default       => "releases/${repo}",
+  }
+  $plugins_repo_path = $repo ? {
+    'stable' => 'plugins/latest',
+    default  => "plugins/${repo}",
```
Comment (domcleal) [inline manifests/install/repos/yum.pp]: Maybe handle `releases/` here too?

**Comment by domcleal** [comment]
Thanks @endyman!

**Pull request closed without merging**